modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectActionLocalServiceTest.java

```diff
@@ -1449,6 +1450,98 @@ public void testUpdateObjectAction() throws Exception {
+				ObjectActionTriggerConstants.KEY_ON_AFTER_ADD,
+				UnicodePropertiesBuilder.put(
+					"secret", "0123456789"
+				).put(
+					"url", "https://onafteradd.com"
+				).build(),
+				true);
+
+		_assertObjectAction(
+			true, "equals(firstName, \"John\")", "Able Description",
+			LocalizedMapUtil.getLocalizedMap("Able Error Message"),
+			LocalizedMapUtil.getLocalizedMap("Able Label"), "Able",
+			systemObjectAction, ObjectActionExecutorConstants.KEY_WEBHOOK,
+			ObjectActionTriggerConstants.KEY_ON_AFTER_ADD,
+			UnicodePropertiesBuilder.put(
+				"secret", "0123456789"
+			).put(
+				"url", "https://onafteradd.com"
+			).build(),
+			ObjectActionConstants.STATUS_NEVER_RAN);
+
+		String liferayMode = SystemProperties.get("liferay.mode");
+
+		SystemProperties.clear("liferay.mode");
+
+		try {
+			systemObjectAction = _objectActionLocalService.updateObjectAction(
+				externalReferenceCode2, systemObjectAction.getObjectActionId(),
+				false, "equals(firstName, \"João\")", "Baker Description",
+				LocalizedMapUtil.getLocalizedMap("Baker Error Message"),
+				LocalizedMapUtil.getLocalizedMap("Baker Label"), "Baker",
+				ObjectActionExecutorConstants.KEY_GROOVY,
+				ObjectActionTriggerConstants.KEY_ON_AFTER_DELETE,
+				UnicodePropertiesBuilder.put(
+					"secret", "30624700"
+				).put(
+					"url", "https://onafterdelete.com"
+				).build());
+		}
+		finally {
+			SystemProperties.set("liferay.mode", liferayMode);
+		}
+
+		_assertObjectAction(
+			true, "equals(firstName, \"John\")", "Able Description",
+			LocalizedMapUtil.getLocalizedMap("Able Error Message"),
+			LocalizedMapUtil.getLocalizedMap("Baker Label"), "Able",
+			systemObjectAction, ObjectActionExecutorConstants.KEY_WEBHOOK,
+			ObjectActionTriggerConstants.KEY_ON_AFTER_ADD,
+			UnicodePropertiesBuilder.put(
+				"secret", "0123456789"
+			).put(
+				"url", "https://onafteradd.com"
+			).build(),
+			ObjectActionConstants.STATUS_NEVER_RAN);
+
+		systemObjectAction = _objectActionLocalService.updateObjectAction(
+			externalReferenceCode2, systemObjectAction.getObjectActionId(),
+			false, "equals(firstName, \"João\")", "Baker Description",
+			LocalizedMapUtil.getLocalizedMap("Baker Error Message"),
+			LocalizedMapUtil.getLocalizedMap("Baker Label"), "Baker",
+			ObjectActionExecutorConstants.KEY_GROOVY,
+			ObjectActionTriggerConstants.KEY_ON_AFTER_DELETE,
+			UnicodePropertiesBuilder.put(
+				"secret", "30624700"
+			).put(
+				"url", "https://onafterdelete.com"
+			).build());
+
+		_assertObjectAction(
+			false, "equals(firstName, \"João\")", "Baker Description",
+			LocalizedMapUtil.getLocalizedMap("Baker Error Message"),
+			LocalizedMapUtil.getLocalizedMap("Baker Label"), "Able",
+			systemObjectAction, ObjectActionExecutorConstants.KEY_GROOVY,
+			ObjectActionTriggerConstants.KEY_ON_AFTER_ADD,
+			UnicodePropertiesBuilder.put(
+				"secret", "30624700"
+			).put(
+				"url", "https://onafterdelete.com"
+			).build(),
+			ObjectActionConstants.STATUS_NEVER_RAN);
```

Comment:
Can we move this scenario after the Add/Assert ?
Then, this test would be:
- successful scenario
- failed scenario 

Like we did with object relationship test. What do you think?